huaweicloud/services/acceptance/evs/data_source_huaweicloud_evs_snapshots_test.go

```diff
@@ -0,0 +1,144 @@
+					resource.TestCheckResourceAttrSet(rName, "snapshots.0.updated_at"),
+					resource.TestCheckResourceAttrSet(rName, "snapshots.0.volume_id"),
+					resource.TestCheckResourceAttrSet(rName, "snapshots.0.service_type"),
+					resource.TestCheckResourceAttrSet(epsIDTestName, "snapshots.#"),
+					resource.TestCheckResourceAttrSet(serviceTypeTestName, "snapshots.#"),
+					resource.TestCheckResourceAttrSet(availabilityZoneTestName, "snapshots.#"),
+					resource.TestCheckResourceAttrSet(dedicatedStorageNameTestName, "snapshots.#"),
+					resource.TestCheckResourceAttrSet(dedicatedStorageIDTestName, "snapshots.#"),
+
+					resource.TestCheckOutput("name_filter_is_useful", "true"),
+
+					resource.TestCheckOutput("status_filter_is_useful", "true"),
+
+					resource.TestCheckOutput("volume_id_filter_is_useful", "true"),
+
+					resource.TestCheckOutput("snapshot_id_filter_is_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDatasourceSnapshots_basic(name string) string {
+	return fmt.Sprintf(`
+%s
+
+data "huaweicloud_evs_snapshots" "test" {
+  depends_on = [huaweicloud_evs_snapshot.test]
+}
+
+data "huaweicloud_evs_snapshots" "enterprise_project_id_filter" {
+  enterprise_project_id = huaweicloud_evs_volume.test.enterprise_project_id
+}
+
+data "huaweicloud_evs_snapshots" "name_filter" {
+  name = data.huaweicloud_evs_snapshots.test.snapshots.0.name
+}
+
+locals {
+  name = data.huaweicloud_evs_snapshots.test.snapshots.0.name
+}
+
+output "name_filter_is_useful" {
+  value = length(data.huaweicloud_evs_snapshots.name_filter.snapshots) > 0 && alltrue(
+    [for v in data.huaweicloud_evs_snapshots.name_filter.snapshots[*].name : v == local.name]
+  )  
+}
+
+data "huaweicloud_evs_snapshots" "status_filter" {
+  status = data.huaweicloud_evs_snapshots.test.snapshots.0.status
+}
+
+locals {
+  status = data.huaweicloud_evs_snapshots.test.snapshots.0.status
+}
+
+output "status_filter_is_useful" {
+  value = length(data.huaweicloud_evs_snapshots.status_filter.snapshots) > 0 && alltrue(
+    [for v in data.huaweicloud_evs_snapshots.status_filter.snapshots[*].status : v == local.status]
+  )  
+}
+
+data "huaweicloud_evs_snapshots" "volume_id_filter" {
+  volume_id = data.huaweicloud_evs_snapshots.test.snapshots.0.volume_id
+}
+
+locals {
+  volume_id = data.huaweicloud_evs_snapshots.test.snapshots.0.volume_id
+}
+
+output "volume_id_filter_is_useful" {
+  value = length(data.huaweicloud_evs_snapshots.volume_id_filter.snapshots) > 0 && alltrue(
+    [for v in data.huaweicloud_evs_snapshots.volume_id_filter.snapshots[*].volume_id : v == local.volume_id]
+  )  
+}
+
+data "huaweicloud_evs_snapshots" "snapshot_id_filter" {
+  snapshot_id = data.huaweicloud_evs_snapshots.test.snapshots.0.id
+}
+
+locals {
+  snapshot_id = data.huaweicloud_evs_snapshots.test.snapshots.0.id
+}
+
+output "snapshot_id_filter_is_useful" {
+  value = length(data.huaweicloud_evs_snapshots.volume_id_filter.snapshots) > 0 && alltrue(
```

Comment:
```suggestion
  value = length(data.huaweicloud_evs_snapshots.snapshot_id_filter.snapshots) > 0 && alltrue(
```
Same question down below.